swap_meet/vendor.py

```diff
@@ -1,2 +1,159 @@
+        other_vendor's inventory
+
+    swap_first_item(friend_vendor):
+        - takes in an instance of another vendor friend_vendor
+        - removes the first item in vendor's inventory and add that to the friend_vendor inventory
+        - removes the first item in friend_vendor inventory and add that to the vendor inventory
+        - returns True
+        - returns False if vendor inventory or friend_vendor inventory is empty 
+
+    get_best_by_category(category):
+        - takes in a string representing the category
+        - looks through vendor's inventory for the item with the highest condition
+        and matching category
+        - returns one item that matches the aforementioned criteria
+
+    swap_best_by_category(other,my_priority,their_priority):
+        - takes in an instance of another vendor (other), a string for a category that the vendor wants
+        to receive (my_priority) and a string for a category that the other vendors wants to receive (other)
+        - passes my_priority and their_priority to the get_best_by_category function to find items with highest
+        condition in each category
+        - passes items to the swap_items to swap those items 
+        - returns True
+        - returns False if my_priority not in vendor's inventory or their priority not in the
+        other's inventory
+
+    swap_by_newest(other):
+        - takes in another instance of vendor (other)
+        - sorts the vendor inventory and other inventory based on the age of items
+        - returns swap_first_item for the vendor and other vendor with sorted inventories
+
+    """
+
+
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError as ve:
+            return False
+
+    def get_by_category(self,category):
+        result = [item for item in self.inventory if item.category == category]
+        return result
+
+    def swap_items(self,other_vendor,my_item,their_item):
+        if not all ((other_vendor, my_item, their_item)):
+            return False
+        if  my_item == their_item:
+            return True
+        elif my_item in self.inventory and their_item in other_vendor.inventory:
+            self.add(their_item)
+            other_vendor.add(my_item)
+            other_vendor.remove(their_item)
+            self.remove(my_item)
+            return True
+
+    def swap_first_item(self, friend_vendor):
+        if not all((friend_vendor.inventory, self.inventory)):
+            return False 
+        
+        friend_vendor.add(self.inventory[0])
+        self.add(friend_vendor.inventory[0])
+        friend_vendor.remove(friend_vendor.inventory[0])
+        self.remove(self.inventory[0])
+        return True
```

Comment:
Nice work! Another option is to use the `swap_items()` as a helper function.

Reply:
Currently, swapping any items has a time complexity of O(n) due to the `remove` method. We can make a subtle improvement to O(1) by using multiple-assignment instead:

```
        self.inventory[0], vendor.inventory[0] = vendor.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple-assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp